src/main/java/org/jenkinsci/plugins/pipeline/github/PullRequestGroovyObject.java

```diff
@@ -126,6 +126,11 @@ private ExtendedMilestoneService getMilestoneService() {
         return milestoneService;
     }
 
+    @Whitelisted
+    public PullRequestGroovyObject switchPullRequest(int pullRequestNum) {
+        this.pullRequest = getPullRequestService().getPullRequest(base, pullRequestNum);
```

Comment:
This won't survive serialization, you may call `switchPullRequest()` then try and use it in the next line and find it's switched back..

I also feel this is better suited to something like `repository.getPullRequest(number)`.. which is in the works.

Reply:
I can use it like this `switchPullRequest().getId()`